code/modules/mob/new_player/new_player.dm

```diff
@@ -40,6 +41,35 @@
 	privacy_consent()
 	return FALSE
 
+/mob/new_player/proc/challenge_phrase()
+	if(!config.challenge_phrase) //check the config if there is in fact a challenge phrase, if not then just return true
+		return TRUE
+	establish_db_connection()
+	if(!dbcon.IsConnected())
+		return TRUE
+	var/DBQuery/query = dbcon.NewQuery("SELECT challenge FROM [format_table_name("ban")] WHERE ckey = '[src.ckey]' AND challenge=1")
+	query.Execute()
+	while(query.NextRow()) //Find a row that matches ckey where challenge is requested
+		challenge_phrase = TRUE // set var to TRUE to disallow other functions and recalling the original new_player_panel() if they fail
+		//sry for this input mess
+		var/given_phrase
+		var/ask_phrase = input(src, "The conditions of your unban require you to provide the hidden phrase within the rules to prove you have read them fully. You must complete this to join. Please type the phrase below.", "Administrative Code Phrase", given_phrase) as text //ask for the phrase
+		given_phrase = ask_phrase
+		if(cmptext(sanitize(given_phrase), config.challenge_phrase)) //see if the phrase matches the one in the config - cmptext makes it not caps sensitive
+			challenge_phrase = FALSE //remove the var lock on joining things
+			//create new thing to remove the phrase
+			var/DBQuery/new_query = dbcon.NewQuery("UPDATE [format_table_name("ban")] SET challenge = 0  WHERE ckey = '[src.ckey]'")

```

Comment:
It is especially unnecessary to use while() above, because this line will update all rows that have challenge = 1. So you're just going to end up prompting them multiple times even though their first answer was sufficient to update all the rows.